### PR TITLE
fix: plugin editor changes do not take effect until run button is clicked

### PR DESCRIPTION
**Pull request by rot1024**
I modified the Plugin Editor to enable testing plugins easily

- Changes do not take effect until run button is clicked
- Smarter layout
- Do refactoring

image

### CODE REVIEW

File: src/components/molecules/PluginEditor/index.tsx [hunk 1]
```diff
@@ -120,139 +58,141 @@ const PluginEditor: React.FC = () => {
...
-                <p>Extension type</p>
-                <select
-                  value={mode}
-                  onChange={e =>
-                    setMode(e.currentTarget.value as "block" | "widget" | "primitive")
-                  }>
-                  <option value="block">Block</option>
-                  <option value="widget">Widget</option>
-                </select>
+                <Icon icon="logo" />
```
Comment (KaWaite) [inline src/components/molecules/PluginEditor/index.tsx:186]: The logo isn't aligned vertically anymore.

File: src/components/molecules/PluginEditor/hooks.ts [hunk 2]
```diff
@@ -37,138 +33,187 @@ const defaultPosition = {
...
 
-  const handleFileReset = useCallback(() => {
+  const handleReset = useCallback(() => {
     if (confirm("Are you sure you want to reset?")) {
-      setSourceCode(hardSourceCode);
-      handleAlignSystemUpdate(widget, defaultPosition);
+      setSourceCode(originalSourceCode);
+      setFileName(originalFileName);
+      setCurrentPosition(defaultPosition);
+      handleRun();
```
Comment (KaWaite) [inline src/components/molecules/PluginEditor/hooks.ts:177]: If I reset once, the WAS and other settings reset, but not the originalSourceCode. I have to do a second reset for it to happen.

Comment (rot1024) [reply src/components/molecules/PluginEditor/hooks.ts:177]: `originalSoruceCode` and `originalFileName` are constants and cannot be changed.

Comment (KaWaite) [reply src/components/molecules/PluginEditor/hooks.ts:176]: Yep I know they are constants. What I mean is it doesn't display the originalSourceCode on first Reset. I have to reset a second time for it to show

Comment (rot1024) [reply src/components/molecules/PluginEditor/hooks.ts]: fixed: [`d674fe9` (#274)](https://github.com/reearth/reearth-web/pull/274/commits/d674fe90aa409db489ade67f7e14fb7f8b1d5873)

File: src/components/molecules/PluginEditor/index.tsx (hunk 1 above)
Comment (rot1024) [reply src/components/molecules/PluginEditor/index.tsx]: No correction will be made this time.

**Review by KaWaite — APPROVED** [review]

**Pull request merged**